ale_linters/rust/rustc.vim

```diff
@@ -0,0 +1,94 @@
+" Author: Daniel Schemala <istjanichtzufassen@gmail.com>
+" Description: rustc for rust files
+
+if !exists('g:ale_rust_ignore_error_codes')
+    let g:ale_rust_ignore_error_codes = []
+endif
+
+
+function! ale_linters#rust#rustc#handle_rustc_errors(buffer_number, errorlines)
+    let l:file_name = fnamemodify(bufname(a:buffer_number), ':t')
+    let l:output = []
+    for l:errorline in a:errorlines
+        " ignore everything that is not Json
+        if l:errorline !~# '^{'
+            continue
+        endif
+
+        let l:error = json_decode(l:errorline)
+
+        if !empty(l:error.code) && index(g:ale_rust_ignore_error_codes, l:error.code.code) > -1
+            continue
+        endif
+
+        for l:span in l:error.spans
+            if l:span.is_primary &&
+                \ (l:span.file_name ==# l:file_name || l:span.file_name ==# '<anon>')
+                call add(l:output, {
+                            \ 'bufnr': a:buffer_number,
+                            \ 'lnum': l:span.line_start,
+                            \ 'vcol': 0,
+                            \ 'col': l:span.byte_start,
+                            \ 'nr': -1,
+                            \ 'text': l:error.message,
+                            \ 'type': toupper(l:error.level[0]),
+                            \ })
```

Comment:
Could you align the continued lines to a new level of indentation like in the rest of the codebase?